src/Orleans/Core/Exceptions.cs

```diff
@@ -29,6 +29,14 @@ protected OrleansException(SerializationInfo info, StreamingContext context)
         }
     }
 
+    public static class OrleansExceptionExtensions
+    {
+        public static bool IsClientNotAvailableException(this OrleansException orleansException)
```

Comment:
Dug into this with Sergey.  We've an approach I'm going to attempt.  Pls see next iteration.